src/components/MessagesList/MessagesList.vue

```diff
@@ -967,6 +967,16 @@ export default {
 			let el = document.getElementById('message_' + this.visualLastReadMessageId)
 			if (el) {
 				el = el.closest('.message')
+				if (el === null || el.offsetParent === null) {
+					// Exception: when the message remains not visible
+					// e.g: it is the last message in collapsed group
+					// unread marker is set to the combined group.
+					// Look for the unread marker itself
+					el = document.querySelector('.new-message-marker')
+					if (el) {
+						el = el.closest('.message')
+					}
```

Comment:
```suggestion
					} else {
						// If we don't have a marker, would it be safe to assume everything is read and return the last message in the list?
						// If it is visible, then we read everything, and `lastReadMessageElement` exists
					}
```

Reply:
> // If we don't have a marker, would it be safe to assume everything is read and return the last message in the list?

The only acceptable case where there is no marker, is that last read message is the last message in the conversation (which is first thing we check before updating read marker). Otherwise, if there is no marker and the last read message is not the last message, it means there is an issue, we should probably log ? 

> // If it is visible, then we read everything, and `lastReadMessageElement` exists

?

Reply:
We probably should

Reply:
We probably should